python_ta/transforms/ExprWrapper.py

```diff
@@ -189,6 +192,97 @@ def parse_container_op(
+            raise Z3ParseException(
+                f"Unhandled binary operation {op} with operator types {left} and {right}."
+            )
+
+    def _parse_number_literal(self, node) -> Optional:
+        """
+        If the subtree from `node` represent a number literal, return the value
+        Otherwise, return None
+        """
+        # positive number
+        if isinstance(node, nodes.Const) and isinstance(node.value, (int, float)):
+            return node.value
+        # negative number
+        elif (
+            isinstance(node, nodes.UnaryOp)
+            and node.op == "-"
+            and isinstance(node.operand, nodes.Const)
+            and isinstance(node.operand.value, (int, float))
+        ):
+            return -node.operand.value
+        else:
+            return None
+
+    def parse_subscript_op(self, node: nodes.Subscript) -> z3.ExprRef:
+        """
+        Convert an astroid Subscript node to z3 expression.
+        This method only supports string values and integer literal (both positive and negative) indexes
+        """
+        value = self.reduce(node.value)
+        if isinstance(value, z3.SeqRef):
+            slice = node.slice
+
+            # handle indexing
+            index = self._parse_number_literal(slice)
+            if isinstance(index, int):
+                return z3.SubString(value, index, index)
+
+            # handle slicing
+            elif isinstance(slice, nodes.Slice):
+                lower = 0 if slice.lower is None else self._parse_number_literal(slice.lower)
+                upper = (
+                    z3.Length(value)
+                    if slice.upper is None
+                    else self._parse_number_literal(slice.upper)
+                )
+                step = 1 if slice.step is None else self._parse_number_literal(slice.step)
+
+                if (
+                    isinstance(lower, int)
+                    and isinstance(upper, (int, z3.ArithRef))
+                    and isinstance(step, int)
+                ):
+
+                    if step == 1:
+                        return z3.SubString(value, lower, upper)
+                    else:
+                        # unhandled case: the upper bound is indeterminant
+                        if upper == z3.Length(value):
+                            raise Z3ParseException(
+                                "Unable to convert a slicing operation with a non-unit step length and an indeterminant upper bound"
+                            )
+
+                        return z3.Concat(
+                            *(z3.SubString(value, i, i) for i in range(lower, upper, step))
+                        )
+                else:
+                    raise Z3ParseException(f"Invalid slice {slice}")
+            else:
+                raise Z3ParseException(f"Invalid index {slice}")
+
+        else:
+            raise Z3ParseException(f"Unhandled subscript operand type {value}")
```

Comment:
Since this is a very asymmetric branch, put this exception check as a guard at the top of the function, so that you don't need to indent the rest.